### PR TITLE
use Awaitility

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

File: tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java [hunk 1]
```diff
@@ -558,13 +560,14 @@ public void close()
...
+                Awaitility.await().atMost(time, unit).until(() -> logs.stream().anyMatch(s -> s.contains(txt)));
+                return true;
+            }
+            catch (ConditionTimeoutException e)
             {
-                boolean result = logs.stream().anyMatch(s -> s.contains(txt));
-                if (result)
-                    return true;
-                Thread.sleep(250);
+                // nothing
```
Comment (sbordet) [inline tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java]: Return `false` from here.

File: tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java [hunk 2]
```diff
@@ -586,20 +589,18 @@ public boolean awaitLogsFileFor(Path logFile, String txt, long time, TimeUnit un
...
-                        return true;
-                    Thread.sleep(250);
-                }
-                return false;
+                Awaitility.await().atMost(time, unit).until(() -> logs.stream().anyMatch(s -> s.contains(txt)));
+                return true;
+            }
+            catch (ConditionTimeoutException e)
+            {
+                // nothing
```
Comment (sbordet) [inline tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java]: Return `false` from here.

File: tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java [hunk 3]
```diff
@@ -558,13 +560,14 @@ public void close()
          */
         public boolean awaitConsoleLogsFor(String txt, long time, TimeUnit unit) throws InterruptedException
         {
-            long end = System.currentTimeMillis() + unit.toMillis(time);
-            while (System.currentTimeMillis() < end)
+            try
+            {
+                Awaitility.await().atMost(time, unit).until(() -> logs.stream().anyMatch(s -> s.contains(txt)));
```
Comment (sbordet) [inline tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java]: Static import `Awaitility.await`.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java (hunk 1 above)
Comment (olamy) [reply tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java]: done

File: tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java (hunk 3 above)
Comment (olamy) [reply tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java]: done

File: tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java (hunk 2 above)
Comment (olamy) [reply tests/jetty-home-tester/src/main/java/org/eclipse/jetty/tests/hometester/JettyHomeTester.java]: done

**Review by sbordet — APPROVED** [review]

**Pull request merged**